### PR TITLE
Fixes #3098 - Add _names for associated resources to API GET methods

### PR DESCRIPTION
**Pull request by mbacovsky**

### CODE REVIEW

File: app/views/api/v1/hostgroups/show.json.rabl [hunk 1]
```diff
@@ -1,3 +1,4 @@
 object @hostgroup
 
 attributes :name, :id, :subnet_id, :operatingsystem_id, :domain_id, :environment_id, :ancestry, :label, :parameters, :puppetclass_ids
+attributes :subnet_name, :operatingsystem_name, :domain_name, :environment_name
```
Comment (domcleal) [inline app/views/api/v1/hostgroups/show.json.rabl:4]: environment might be better as a child, to match the host/show view

**Comment by domcleal** [comment]
There are a few test failures, can you look into those please @mbacovsky?

I noticed the following other views where they could be added:
- app/views/api/v2/taxonomies/show.json.rabl
- app/views/api/v1/bookmarks/show.json.rabl
- app/views/api/v1/architectures/show.json.rabl
- app/views/api/v1/users/index.json.rabl
- app/views/api/v1/media/show.json.rabl
- app/views/api/v1/domains/show.json.rabl
- app/views/api/v1/subnets/index, show.json.rabl

Does that make sense?

**Comment by domcleal** [comment]
@isratrade would you mind reviewing please?

**Comment by isratrade** [comment]
@mbacovsky, I prefer to have _name attributes on v2 rabl templates only and leave v1 as is. My goal is to make the response configurable to generate 
- :abc_id, :abc_name in parent node
- or child note 'abc': {'id': 3, 'name': 'abcdef'}

**Comment by mbacovsky** [comment]
@isratrade adding _names to just v2 would duplicate many of the rabl files that I wanted to avoid. But it make more sens with the configurable response. If I recall it correctly, it was almost ready as rabl dsl extension in one of your earlier PRs. I liked that approach so I'd like to revive the relevant parts and go on that. What do you think?

**Comment by isratrade** [comment]
@mbacovsky, in any case I plan to add v2 controllers and v2 rabl templates for all controllers and not inherit from v1. So, yes, there will be duplication, but at some point, I assume we will discontinue v1 in future releases.
Regarding the rabl dsl extension, it was PR #844, but PR #850 was merged instead.  I don't think it makes sense to add #844

**Comment by mbacovsky** [comment]
@isratrade, thanks for explanation. My idea with PR #844 was to keep the rabl syntax and use associated names to encapsulate the configurable behaviour:

```
associated_attributes :environment, :model
```

would render either two parent nodes for each or child node depending on configuration.
Would that make sense?

**Comment by ohadlevy** [comment]
@isratrade ping ?

**Comment by isratrade** [comment]
@ohadlevy, @mbacovsky, several PR's including this one seem to be waiting on issue #3017.  I optimistically hope to have a PR for this by end of the day tomorrow, and then these changes can be implemented on the v2 rabl views.

**Comment by mbacovsky** [comment]
Seems this is re-implemented in #844. I'm going to cancel this one...

**Comment by isratrade** [comment]
@mbacovsky, yes you can close this one.   I changed only on the v2 json rabl templates.

**Pull request closed without merging**